java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -1021,7 +1074,19 @@ private void setNullAt(int index) {
     }
 
     public final ColumnBuilder appendNull() {
-      growBuffersAndRows(true, 0);
+      // Increments row number. And update offsets and data buffer along with the valid buffer.
+      // NOTE: The growth of valid buffer must happen after rowCapacity updated.
+      if (type.typeId == DType.DTypeEnum.LIST) {
```

Comment:
I am not sure which is more expensive virtual method calls or a large if/else statement like this.  It would be nice to see some performance testing to see if we should have a lambda function that is set based off of the DType in the constructor to grow the buffers, or if it is better to keep this as is.

Reply:
I replaced the if/else statements in `appendNull` with a Runnable instance: `nullHandler`. So, the type dispatching only happens when we create the ColumnBuilder. I tested these two approaches locally. The new approach outperformed by a tiny margin:
```
The result of new approach:
the cost of the 0 run: 17897
the cost of the 1 run: 18890
the cost of the 2 run: 15329
the cost of the 3 run: 15203
the cost of the 4 run: 15677
the cost of the 5 run: 18500
the cost of the 6 run: 16128
the cost of the 7 run: 16483
the cost of the 8 run: 15321
the cost of the 9 run: 15694
the cost of the 10 run: 17039
the cost of the 11 run: 16420
the cost of the 12 run: 14998
the cost of the 13 run: 15068
the cost of the 14 run: 15115
the cost of the 15 run: 15137
the cost of the 16 run: 15150
the cost of the 17 run: 15133
the cost of the 18 run: 14991
the cost of the 19 run: 15055
average cost of 20 runs: 15961.4

the cost of the 0 run: 18801
the cost of the 1 run: 16943
the cost of the 2 run: 16918
the cost of the 3 run: 16776
the cost of the 4 run: 16804
the cost of the 5 run: 16747
the cost of the 6 run: 17783
the cost of the 7 run: 16869
the cost of the 8 run: 16762
the cost of the 9 run: 19127
the cost of the 10 run: 17242
the cost of the 11 run: 17303
the cost of the 12 run: 16799
the cost of the 13 run: 16724
the cost of the 14 run: 16802
the cost of the 15 run: 16759
the cost of the 16 run: 16709
the cost of the 17 run: 16561
the cost of the 18 run: 16627
the cost of the 19 run: 16520
average cost of 20 runs: 17078.8
```
Here is the test snippet:
```
    int runs = 20;
    int estimatedRows = 1024;
    int totalRows = Integer.MAX_VALUE - 10;
    long duration;
    HostColumnVector.DataType type = new HostColumnVector.BasicType(true, DType.INT64);
    duration = 0L;
    for (int x = 0; x < runs; x++) {
      long start = System.currentTimeMillis();
      try (HostColumnVector.ColumnBuilder cb = new HostColumnVector.ColumnBuilder(
          type, estimatedRows)) {
        for (int i = 0; i < totalRows; i++) {
          cb.appendNull();
        }
      }
      long inc = System.currentTimeMillis() - start;
      System.out.println("the cost of the " + x + " run: " + inc);
      duration += inc;
    }
    System.out.println("average cost of " + runs + " runs: " + (float) duration / runs);
```